src/ngMock/angular-mocks.js

```diff
@@ -1481,8 +1483,9 @@ function createHttpBackendMock($rootScope, $timeout, $delegate, $browser) {
    *      ```
    *    – The respond method takes a set of static data to be returned or a function that can
    *    return an array containing response status (number), response data (Array|Object|string),
-   *    response headers (Object), and the text for the status (string). The respond method returns
-   *    the `requestHandler` object for possible overrides.
+   *    response headers (Object), HTTP status text (string), and XMLHttpRequest status (string:
```

Comment:
It doesn't get it. It returns it. See http://jsbin.com/vumefiz/edit?html,js,output

Reply:
Sorry, I misread that 😳

Reply:
Actually, our [createResponse()][1] method isn't consistently treating `XHR status`. It does not accept one when passing the data directly to `.respond()` (and it automatically uses `complete`as the XHR status), but it does accept one when a function is passed to `createResponse()`.

I think this is a bug (or at least unintentional). `createResponse()` should look something like:

```diff
 function createResponse(status, data, headers, statusText) {
-  if (angular.isFunction(status)) return status;
+  if (angular.isFunction(status)) {
+    return function(method, url, data, headers, params) {
+      var response = status(method, url, data, headers, params);
+      response[4] = 'complete';
+      return response;
+    };
+  }

   return function() {
     return angular.isNumber(status)
         ? [status, data, headers, statusText, 'complete']
         : [200, status, data, headers, 'complete'];
   };
 }
```

But this is unrelated to this PR and would probably be a breaking change now 😞 

[1]: https://github.com/thorn0/angular.js/blob/04024ed13437b6e8ffca2158625f006ee6cfc8cc/src/ngMock/angular-mocks.js#L1355-L1363

Reply:
Why do you think so? What if we want to test the behavior of our app with XHR statuses other than `complete` (e.g. `error`)?

Reply:
Because the XHR status is not something that can be arbitrarily specified. It is tied to the way the XHR complete. You can't have `error` with a "normal" response (it is something that will not come up in the actual app).

Reply:
I.e., the only valid XHR status when returning a response (via the function returned by `createResponse()`) is `complete`.

Reply:
But (like I said), this is not something to worry about in this PR. It is not related (and that ship has probably sailed :grin:).

Reply:
> You can't have error with a "normal" response (it is something that will not come up in the actual app

I believe I can. It happens in case of network error, doesn't it?

Reply:
Yes, that's not a "normal" response. I.e. an XHR status other than `complete` can only happen if the XHR's `.onabort`/`.onerror`/`.ontimeout` is called. In that case, you would have a status of `0` and empty values for other fields (`statusText`, `headers`, etc).

But the mock `$httpBackend` is not really designed to let you test these scenarios. Sure you can emulate it by returning the correct values from your response function, you need to know `$httpBackend` internals. E.g. returning `[200, 'OK', ..., 'aborted']` is a scenario that will never happen in the actual app and will lead to unexpected results (because `aborted` will have been called with a "success" status).

So, ideally, we should not let users specify the XHR status, but instead infer it from other params.
Even more ideally, `$httpBackend` could support testing scenarions when an XHR errored, or timed out or was aborted (and again automatically setting the correct XHR status; not letting the user specify it).

But as it is now, `$httpBackend` is better used with "normal" responses (unless you really know what you are doing) 😛

Reply:
In a separate PR, we can add simple checks and throw in cases like `[200, 'OK', ..., 'aborted']`. What are your thoughts on that?

Reply:
I wouldn't be opposed to that :smiley: